app/src/lib/components/calender.svelte

```diff
@@ -0,0 +1,117 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import CalenderItem from '$lib/components/CalenderItem.svelte';
+
+	interface EventData {
+		title: string;
+		sport: string;
+		startTime: string;
+		endTime: string;
+		location: string;
+	}
+	interface EventDataWithPos extends EventData {
+		colStart: number;
+		rowStart: number;
+		rowSpan: number;
+	}
+
+	// calender is between {calenderStartHour}th hour and {calenderEndHour}th hour
+	let {
+		events,
+		calenderStartHour = 6,
+		calenderEndHour = 24
+	}: { calenderStartHour?: number; calenderEndHour?: number; events: EventData[] } = $props();
+
+	// removing {calenderStartHour} hours from total day of 24 hours
+	let rows = calenderEndHour - calenderStartHour;
+	let cols = $state(1);
+	let eventsWithPos: EventDataWithPos[] = $state([]);
+	let occupiedGrids: number[][] = [[]];
+
+	onMount(() => {
+		eventsWithPos = events.map((event) => {
+			return { ...event, ...calculatePos(event) };
+		});
+		cols = occupiedGrids.length;
+	});
+
+	// returns the position of a specific event on the event grid
+	function calculatePos(event: EventData) {
+		let startRow = getRow(new Date(event.startTime));
+		let endRow = getRow(new Date(event.endTime));
+
+		// finding column with no clashes
+		let col = -1;
+		for (let i = 0; i < occupiedGrids.length; i++) {
+			if (!isAnyNumberInRange(startRow, endRow - 1, occupiedGrids[i])) {
+				col = i;
+				break;
+			}
+		}
+		if (col === -1) {
+			col = occupiedGrids.length;
+			occupiedGrids.push([]);
+		}
+		occupiedGrids[col].push(...Array.from({ length: endRow - startRow }, (_, i) => startRow + i));
+
+		return {
+			colStart: col + 1,
+			rowStart: startRow,
+			rowSpan: endRow - startRow
+		};
+	}
+	// converts hours and mins to its corresponding row
+	function getRow(date: Date) {
+		return (date.getUTCHours() - calenderStartHour) * 4 + 1 + date.getUTCMinutes() / 15;
+	}
+	// finds if any number between x & y (inclusive) is in arr
+	function isAnyNumberInRange(x: number, y: number, arr: number[]) {
+		return arr.some((num) => num >= x && num <= y);
+	}
+</script>
+
+<div class="flex flex-row bg-transparent p-2 text-white">
+	<!-- Time stamps -->
+	<div class="flex flex-col text-xs">
+		{#each { length: rows }, i}
+			<div class="h-20 w-10 pr-1 text-right">
+				{((i + calenderStartHour - 1) % 12) + 1}
+				{i < 12 - calenderStartHour ? 'AM' : 'PM'}
+			</div>
+		{/each}
+		<div class="w-10 pr-1 text-right">

```

Comment:
Here is a possible offender to the differing padding / margins applied on the event view

Reply:
The actual cause was this line:

https://github.com/crux-bphc/arena-app/blob/bd7bdc1d3323f55e68ef89b7e1cf22a06629e938/app/src/lib/components/CalenderItem.svelte#L15

It's fixed in 2e9b891.